const catchError = require('../utils/catchError');
const Purchase = require('../models/Purchase');
const Product = require('../models/Product');
const Cart = require('../models/Cart');
const Image = require('../models/Image');

const getAll = catchError(async(req, res) => {
    const results = await Purchase.findAll({
        where: {userId: req.user.id},
        include: [{
            model: Product,
            include:[Image]
        }],
    });
    return res.json(results);
});

const create = catchError(async(req, res) => {
    const userId = req.user.id
    const carts = await Cart.findAll({
        where:{userId},
        attributes:['userId', 'productId', 'quantity'],
        raw: true
    });
    const purchases = await Purchase.bulkCreate(carts);
    await Cart.destroy({where:{userId}});
    return res.status(201).json(purchases);
});

module.exports = {
    getAll,
    create
}